packages/react-button/src/components/Button/Button.types.ts

```diff
@@ -1,175 +1,121 @@
-     * important to keep a consistent tab order for screen reader and keyboard users.
-     * @defaultvalue false
-     */
-    disabledFocusable?: boolean;
+    // /**
+    // eslint-disable-next-line @fluentui/max-len
+    //  * When set, allows the button to be focusable even when it has been disabled. This is used in scenarios where it is
+    //  * important to keep a consistent tab order for screen reader and keyboard users.
+    //  * @defaultvalue false
+    //  */
+    // disabledFocusable?: boolean;
 
-    /** A button can fill the width of its container. */
-    block?: boolean;
+    // /** A button can fill the width of its container. */
+    // block?: boolean;
 
     /** A button can contain only an icon. */
     iconOnly?: boolean;
 
     /** An icon button can format its icon to appear before or after its content. */
     iconPosition?: 'before' | 'after';
 
-    /** A button that inherits its background and has a subtle appearance. */
-    inverted?: boolean;
-
-    /** Opt in to a named variant which can be supplied by a theme. */
-    variant?: string;
-
-    /** A button can show a loading indicator. */
-    loading?: boolean;
-
-    /**
-     * Called after a user clicks the button.
-     * @param event - React's original SyntheticEvent.
-     * @param data - All props.
-     */
-    // onClick?: ComponentEventHandler<ButtonProps>;
-
-    /**
-     * Called after a user focuses the button.
-     * @param event - React's original SyntheticEvent.
-     * @param data - All props.
-     */
-    // onFocus?: ComponentEventHandler<ButtonProps>;
+    // /** A button can show a loading indicator. */
+    // loading?: boolean;
 
     /** A button can emphasize that it represents the primary action. */
     primary?: boolean;
 
-    /** A button can emphasize that it represents an alternative action. */
-    secondary?: boolean;
+    // /** A button can blend into its background to become less emphasized. */
+    // subtle?: boolean;
 
-    /** A button can blend into its background to become less emphasized. */
-    subtle?: boolean;
-
-    /** A button can have no background styling and just be emphasized through its content styling. */
-    transparent?: boolean;
+    // /** A button can have no background styling and just be emphasized through its content styling. */
+    // transparent?: boolean;
 
     /** A button can be sized. */
     size?: SizeValue;
-
-    // TODO: Deprecate or rename to textOnly for alignment with iconOnly?
-    /** A button can be formatted to show only text in order to indicate a less-pronounced action. */
-    // text?: boolean;
-
-    tokens?: RecursivePartial<ButtonTokens>;
   };
 
 /**
  * {@docCategory Button}
  */
 export interface ButtonState extends ButtonProps {
-  'data-is-focusable': boolean;
   ref: React.RefObject<HTMLButtonElement>;
+
+  icon?: ObjectShorthandProps<React.HTMLAttributes<HTMLSpanElement>>;
+  children?: ObjectShorthandProps<React.HTMLAttributes<HTMLSpanElement>>;
```

Comment:
Are you going to keep it as it is? Or is this a temporary typing?

Reply:
Temporary "fix" until we truly fix the getSlots implementation. There are TODOs added to the code about this in various places. Button should be the last component we merge for preview, we'll double back and fix hooks immediately following.